tests/test_pca.py

```diff
@@ -180,48 +181,41 @@ def test_pca_warnings(array_type, zero_center, pca_params):
         raise
 
 
-# This warning test is out of the fixture because it is a special case in the logic of the function
-def test_pca_warnings_sparse():
-    for array_type in (sparse.csr_matrix, sparse.csc_matrix):
-        A = array_type(A_list).astype("float32")
-        adata = AnnData(A)
-        with pytest.warns(UserWarning, match="not work with sparse input"):
-            sc.pp.pca(adata, svd_solver="randomized", zero_center=True)
-
-
 def test_pca_transform(array_type):
-    A = array_type(A_list).astype("float32")
+    adata = AnnData(array_type(A_list).astype("float32"))
     A_pca_abs = np.abs(A_pca)
-    A_svd_abs = np.abs(A_svd)
-
-    adata = AnnData(A)
 
     with warnings.catch_warnings(record=True) as record:
         sc.pp.pca(adata, n_comps=4, zero_center=True, dtype="float64")
     assert len(record) == 0, record
 
     assert np.linalg.norm(A_pca_abs[:, :4] - np.abs(adata.obsm["X_pca"])) < 2e-05
 
-    with warnings.catch_warnings(record=True) as record:
+
+def test_pca_transform_warning_randomized(array_type):
+    adata = AnnData(array_type(A_list).astype("float32"))
+    A_pca_abs = np.abs(A_pca)
+
+    with (
+        pytest.warns(UserWarning, match="Ignoring.*'randomized'")
+        if sparse.issparse(adata.X)
+        else nullcontext()
+    ):
         sc.pp.pca(
             adata,
-            n_comps=5,
+            n_comps=4,
```

Comment:
this is iffy, the `_pca_with_sparse` was able to do 5 earlier. probably need to fix this.

Reply:
actually that was the backwards compat code I eliminated, so it’s fine.